picard/file.py

```diff
@@ -412,6 +413,10 @@ def _save_and_rename(self, old_filename, metadata):
         return new_filename
 
     def _saving_finished(self, result=None, error=None):
+        if error is None and result is not None:
+            new_filename = result
+            # Add new path info after successful rename
+            self.metadata["~path_saved"] = new_filename
```

Comment:
I must be missing something because it appears that the `"~path_saved"` variable is being set after the file has been saved.  Can you provide an example of how this would be used?

Reply:
Now that Bob has prompted me to think further, it seems to me that we need to know:

1. The current path i.e. where the file actually is - displayed in Old metadata.
2. The future path if we were to save the file now - displayed in New metadata.
3. The previous path if we have saved the file since it has been loaded.

To make life easiest for scripters, once a file is loaded and before it is saved we need to decide whether the previous path is set to the current path or left empty / non-existent.

And we should decide whether to save the entire path (directory path and filename as a string) or the directory path and the filename separately or possibly both. (Scripters may want directory and filename separate or they may want it together. UI needs them separate.)

Reply:
> 3\. The previous path if we have saved the file since it has been loaded.

Do we need this?  When the file is saved, I think the file variables are automatically updated to the new path and filename, so I'm not sure what the use case would be for the original path and filename.

> And we should decide whether to save the entire path (directory path and filename as a string) **_or_** the directory path and the filename separately **_or_** possibly both.

My suggestion is to keep the path and filename as separate variables for the maximum flexibility of use.  In Picard's scripting, it's easier to combine the two than split them out from a combined value.

Another consideration is how path separators are displayed and saved.  The current file naming script preview displays the drive and path as used for the file system (using a colon and backslashes for windows paths).  I think this is fine (and appropriate) for the display in the table in the Metadata pane.  It's probably okay for the values stored in the variables as well, but we may need to be careful because of the use of the backslashes in paths on windows systems.  I assume they would need to be escaped for the tags/variables, although I really don't know the details of how they are saved in the file metadata (if the user chooses to copy the variable to a tag so it is saved).

> **Motivation:** We should try to get this exactly right from the start before we create a backwards compatibility problem for scripters.

I agree, although I don't think we're actually changing any of the existing tags/variables so we **shouldn't** have any backwards compatibility issues.  [Famous last words...  😜]